### PR TITLE
[Mage] From the Ashes, Arcane Orb, & General Cleanup

### LINKED ISSUE #3948: [Arcane Mage] Arcane Orb multiple target
The tooltip of the "Use your talents effectively" category says for Arcane Orb Avg. Hits Per Cast: "Arcane Orb is primarily an AoE Spell, so you should only choose it on fights with multiple targets [...]"
Well, the whole talent row doesn't have any single target spells. Reverberate ist useless in pure single target, and Supernova is in all practical purposes less damage than Arcane Orb. So yeah, Arcane Orb *is* the singletarget choice. ^^

### PR DESCRIPTION
**Pull request by Sharrq**
- Added the From the Ashes Talent for Fire (Towards #3657 )
- Added Covenant Abilities to the Spellbook for all 3 specs
- Removed the GCD for Icy Veins and Arcane Power
- Adjusted Arcane Orb since it is acceptable for ST now (Fixes #3948 )
- Adjusted Conduit data for the latest Beta Build (Towards #3675 )
- Removed types from the implicitly defined variables in all specs

I also undid some of what i did with constants and have gone the route of using a constants file for the whole class instead of individual spec constants. Additionally, if a constant is only used in one module, I removed it from the centralized constants file and put it back into the relevant module.

### CODE REVIEW

File: src/parser/mage/arcane/modules/talents/ArcaneOrb.tsx [hunk 1]
```diff
@@ -77,11 +77,11 @@ class ArcaneOrb extends Analyzer {
...
+        tooltip={`You averaged ${formatNumber(this.averageHitsPerCast)} hits per cast of Arcane Orb. ${this.badCasts > 0 ? `Additionally, you cast Arcane Orb ${this.badCasts} times without hitting anything.` : '' } Casting Arcane Orb when it will only hit one target is still beneficial and acceptable, but if you can aim it so that it hits multiple enemies then you should.`}
       >
-        <BoringSpellValueText spell={SPELLS.ARCANE_FAMILIAR_TALENT}>
+        <BoringSpellValueText spell={SPELLS.ARCANE_ORB_TALENT}>
           <>
-            {formatNumber(this.averageHitPerCast)} <small>Average hits per cast</small>
+            {formatNumber(this.averageHitsPerCast)} <small>Average hits per cast</small>
           </>
```
Comment (Pewtro) [inline src/parser/mage/arcane/modules/talents/ArcaneOrb.tsx:85]: If you wanted to you could use the AverageTargetsHit helper here

File: src/parser/mage/fire/modules/talents/FromTheAshes.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(FIRE_DIRECT_DAMAGE_SPELLS), this.onCritDamage);
+  }
+
+  //Look for crit damage events to reduce the cooldown on Kindling
+  onCritDamage(event: DamageEvent) {
+    if (!this.spellUsable.isOnCooldown(SPELLS.PHOENIX_FLAMES.id) || event.hitType !== HIT_TYPES.CRIT) {
+      return;
+    }
+    const chargesOnCD = PHOENIX_FLAMES_MAX_CHARGES - this.spellUsable.chargesAvailable(SPELLS.PHOENIX_FLAMES.id);
+    this.cooldownReduction += this.spellUsable.reduceCooldown(SPELLS.PHOENIX_FLAMES, MS_REDUCTION_PER_STACK * chargesOnCD);
```
Comment (Pewtro) [inline src/parser/mage/fire/modules/talents/FromTheAshes.tsx]: This should be chargesOffCD, right?

**Review by Pewtro — COMMENTED** [review]
1 possibility for now or for future (re: avg targets)
1 clarifying variable name question

Nothing blocking.

File: src/parser/mage/fire/modules/talents/FromTheAshes.tsx (hunk 2 above)
Comment (Sharrq) [reply src/parser/mage/fire/modules/talents/FromTheAshes.tsx]: No. From the Ashes gives you CDR for every charge that is on CD. so this is the max charges (2) - the number available

Comment (Pewtro) [reply src/parser/mage/fire/modules/talents/FromTheAshes.tsx]: image
I just looked it up on wowhead and it says off cooldown?

Comment (Sharrq) [reply src/parser/mage/fire/modules/talents/FromTheAshes.tsx]: woops, i misread it

Comment (Pewtro) [reply src/parser/mage/fire/modules/talents/FromTheAshes.tsx]: That said - is it really per off CD charge for the CDR, and not only for the mastery? Blizzard tooltips fine at work here 😄

Comment (Sharrq) [reply src/parser/mage/fire/modules/talents/FromTheAshes.tsx]: actually, re-reading it again, i suppose its 1 sec CDR regardless of charges, and the mastery is based on charges

**Comment by emallson** [comment]
@Sharrq looks like you handled the requested changes. we good to merge (after conflict resolution)?

**Comment by Sharrq** [comment]
Yea I'm not home right now but my plan is to merge this and then if I change the average hits thing I'll do it for arcane orb and frozen orb in a separate pr

**Pull request merged**